### PR TITLE
Rename frames

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Renames the frames as discussed with @kidder @wthrowe @nilsdeppe (taken over from #1753):

- [x] `Frame::Logical` -> `Frame::ElementLogical` (because we are adding `Frame::BlockLogical`)
- [ ] `Frame::Grid` -> `Frame::GlobalTimeIndependent` (to reflect these are the coordinates that cover the entire domain but have no time-dependent mappings, so can be used for caching)
- [ ] `Frame::Inertial` -> `Frame::System` (this is the frame the system equations are formulated in, i.e. the system coordinates. Most tensors will be in this frame.)

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Please fill out the `Proposed changes` section to contain a summary of what is actually being changed.

**Comment by nilsvu** [comment]
@nilsdeppe done. Do you want to get in #1777 quickly so only one of us has to rebase? :)

**Comment by nilsdeppe** [comment]
Sure :) 

One question I have about this PR: the diff line count is different from what I had. Do you have an idea of what I missed or why this is?

**Comment by nilsvu** [comment]
@nilsdeppe I think it's because `physical` is the same length as `inertial`, but `system` is slightly shorter, so more code got reformatted.

**Comment by nilsvu** [comment]
Since this is running clang-tidy over a lot of files, it turned up two clang-tidy warnings that were probably ignored when they were merged. One is in `boost/math/tools/toms748_solve.hpp` and the other in `ValenciaDivClean/TagsDeclarations.hpp`. Any suggestions what to do with those?

```
/usr/include/boost/math/tools/toms748_solve.hpp:209:35: error: The left operand of '-' is a garbage value [clang-analyzer-core.UndefinedBinaryOperatorResult,-warnings-as-errors]
Suppressed 50350 warnings (50095 in non-user code, 255 NOLINT).
   T A = safe_div(T(fd - fb), T(d - b), tools::max_value<T>());
```

```
/__w/spectre/spectre/src/Evolution/Systems/GrMhd/ValenciaDivClean/TagsDeclarations.hpp:16:8: error: no definition found for 'CharacteristicSpeeds', but a definition with the same name 'CharacteristicSpeeds' found in another namespace 'RadiationTransport::M1Grey::Tags' [misc-forward-declaration-namespace,-warnings-as-errors]
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
struct CharacteristicSpeeds;
1 warning treated as error
       ^
/__w/spectre/spectre/src/Evolution/Systems/RadiationTransport/M1Grey/Tags.hpp:24:8: note: a definition of 'CharacteristicSpeeds' is found here
struct CharacteristicSpeeds : db::SimpleTag {
       ^
```

**Comment by kidder** [comment]
you can ignore the first one (about boost code); it is a warning that we (@wthrowe @kidder @nilsdeppe) have never been able to fix or tell clang-tidy to ignore.

File: src/DataStructures/Tensor/IndexType.hpp [hunk 1]
```diff
@@ -117,8 +117,8 @@ inline std::string prefix<Frame::GlobalTimeIndependent>() noexcept {
 }
 
 template <>
-inline std::string prefix<Frame::Inertial>() noexcept {
-  return "";
+inline std::string prefix<Frame::System>() noexcept {
+  return "System_";
```
Comment (wthrowe) [inline src/DataStructures/Tensor/IndexType.hpp:175]: If you're changing the behavior here, change the documentation above to match.  (And really changes like this should be in a separate commit.)

File: src/DataStructures/Tensor/IndexType.hpp [hunk 2]
```diff
@@ -79,6 +79,14 @@ struct GlobalTimeIndependent {};
  */
 struct System : FrameIsPhysical {};
 
+/// \cond
+// Deprecated frame names with aliases to their replacements
+using Logical[[deprecated("Use `Frame::ElementLogical`")]] = ElementLogical;
+using Grid[[deprecated("Use `Frame::GlobalTimeIndependent`")]] =
+    GlobalTimeIndependent;
+using Inertial[[deprecated("Use `Frame::System`")]] = System;
+/// \cond
```
Comment (wthrowe) [inline src/DataStructures/Tensor/IndexType.hpp:88]: `\endcond`

**Review by wthrowe — CHANGES_REQUESTED** [review]
When this is done, file an issue for us to figure out where "ElementLogical" should be changed to "BlockLogical".

File: src/DataStructures/Tensor/IndexType.hpp (hunk 1 above)
Comment (nilsdeppe) [reply src/DataStructures/Tensor/IndexType.hpp:175]: I'd argue even a separate PR... A major rename like this should only do that rename and nothing else to keep things as simple as possible.

**Pull request closed without merging**